### ISSUE TITLE
AttributeError

### ISSUE THREAD

**Issue by jp1151419** [DIGITALCRIMINAL/OnlyFans#789]
> Scrape Processing
> Name: ston3dlg
> Type: Profile
> Type: Stories
> Scraping [photo]. Should take less than a minute.
> Scraping [photo]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Scraping [photo]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Processing metadata.
> Finished processing metadata.
> INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
> INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
> INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
> INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
> Renaming files.
> Type: Posts
> Scrape Attempt: 1/100
> Found: Posts
> Scraping [photo]. Should take less than a minute.
> Scraping [photo]. Should take less than a minute.
> Scraping [photo]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [photo]. Should take less than a minute.
> Scraping [photo]. Should take less than a minute.
> Scraping [photo]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [video | stream | gif]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Scraping [audio]. Should take less than a minute.
> Scraping [text]. Should take less than a minute.
> Processing metadata.
> Finished processing metadata.
> INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
> INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
> INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
> INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
> Renaming files.
> Type: Archived
> Traceback (most recent call last):
>   File "X:\OnlyFans\start_ofd.py", line 40, in <module>
>     main_datascraper.start_datascraper(json_config, site_name_lower)
>   File "X:\OnlyFans\datascraper\main_datascraper.py", line 87, in start_datascraper
>     names = main_helper.process_names(
>   File "X:\OnlyFans\helpers\main_helper.py", line 602, in process_names
>     result = module.start_datascraper(
>   File "X:\OnlyFans\modules\onlyfans.py", line 155, in start_datascraper
>     results = prepare_scraper(
>   File "X:\OnlyFans\modules\onlyfans.py", line 680, in prepare_scraper
>     master_set = subscription.get_archived(api)
>   File "X:\OnlyFans\apis\onlyfans\onlyfans.py", line 441, in get_archived
>     item["results"] = self.get_archived_posts()
>   File "X:\OnlyFans\apis\onlyfans\onlyfans.py", line 428, in get_archived_posts
>     self.scraped.Archived.Posts = results
> AttributeError: 'content_types' object has no attribute 'Archived'

**Comment by jp1151419** [comment]
I get that error here and there, the account on which it happens is never the same. Because of this error I'm having a hard time getting the script to run more than once every couple days. Could it be due to a bug with the website? They sure have some that come and go..

To give you an idea the account I'm trying to rip with has approx 600 subscriptions so run do take a while to complete. A single error like this brings me back a couple hours each time.

**Comment by jp1151419** [comment]
I was able to scrape that same account on it's own but it stopped the scrapper three times in a row when doing 0=all

**Comment by DIGITALCRIMINAL** [comment]
Are you using the latest commit?

**Comment by jp1151419** [comment]
No, last run was with b54f902. I didn't see that commit that came out 1h before my post, my bad.

**Comment by jp1151419** [comment]
Just tried it, no success

> Scrape Processing
Name: babylaceylea
Type: Profile
Type: Stories
No Stories Found.
Traceback (most recent call last):
  File "X:\OnlyFans\start_ofd.py", line 40, in <module>
    apis = main_datascraper.start_datascraper(json_config, site_name_lower)
  File "X:\OnlyFans\datascraper\main_datascraper.py", line 116, in start_datascraper
    names = main_helper.process_names(
  File "X:\OnlyFans\helpers\main_helper.py", line 603, in process_names
    result = module.start_datascraper(
  File "X:\OnlyFans\modules\onlyfans.py", line 155, in start_datascraper
    results = prepare_scraper(
  File "X:\OnlyFans\modules\onlyfans.py", line 724, in prepare_scraper
    delattr(subscription.scraped, api_type)
AttributeError: Stories

**Comment by jp1151419** [comment]
Multiple profiles are having issues, resulting in either AttributeError: Stories or Archives

I'm blacklisting them as they happen but each time I have to restart and some of those I haven't gotten to scrape yet, which I want to do before they expire.

**Comment by jp1151419** [comment]
@DIGITALCRIMINAL issue persists with 387da4d

**Comment by jp1151419** [comment]
Scraping [video | stream | gif]. Should take less than a minute.
Scraping [audio]. Should take less than a minute.
Scraping [text]. Should take less than a minute.
Processing metadata.
Finished processing metadata.
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
Renaming files.
Type: Archived
Traceback (most recent call last):
  File "X:\OnlyFans\start_ofd.py", line 43, in <module>
    apis = main_datascraper.start_datascraper(json_config, site_name_lower)
  File "X:\OnlyFans\datascraper\main_datascraper.py", line 98, in start_datascraper
    names = main_helper.process_names(
  File "X:\OnlyFans\helpers\main_helper.py", line 602, in process_names
    result = module.start_datascraper(
  File "X:\OnlyFans\modules\onlyfans.py", line 145, in start_datascraper
    results = prepare_scraper(
  File "X:\OnlyFans\modules\onlyfans.py", line 674, in prepare_scraper
    master_set = subscription.get_archived(api)
  File "X:\OnlyFans\apis\onlyfans\onlyfans.py", line 457, in get_archived
    item["results"] = self.get_archived_posts()
  File "X:\OnlyFans\apis\onlyfans\onlyfans.py", line 444, in get_archived_posts
    self.scraped.Archived.Posts = results
AttributeError: 'content_types' object has no attribute 'Archived'

This time it was on showmeyourtitch using the last build, 63f63e6. It doesn't happen if I start with a new .sites output folder though.

**Comment by DIGITALCRIMINAL** [comment]
Hoping I can fix this in a few hours.

**Comment by jp1151419** [comment]
That would be great, I have to correct my last post; Nothing I can do on my end helps. Neither starting fresh with new Metadata nor a new folder altogether.

**Comment by DIGITALCRIMINAL** [comment]
You've got her archived posts metadata?

**Comment by jp1151419** [comment]
No I don't, at least a recent copy of it

**Comment by DIGITALCRIMINAL** [comment]
Send it to me via email, please. I can't fix the problem unless I got it.

**Comment by jp1151419** [comment]
isn't my 2fa key in that file?

**Comment by DIGITALCRIMINAL** [comment]
> I have a copy from one week ago

What about ston3dlg?

**Comment by jp1151419** [comment]
nope, don't have any archived post metadata for her, nor any of babylaceylea's story post metadata

just so you know these last two weeks I've had accounts I couldn't scrape that I used to be able without an issue. On at least on occasion I was able to scrape of the problematic accounts by doing them separately and with an empty .sites folder

**Comment by DIGITALCRIMINAL** [comment]
I was able to scrape multiple accounts without any issues :/

**Comment by jp1151419** [comment]
So I don't know if something changed on OF's side but I'm still rocking build 63f63e6 and it now works. I first got it working by scraping only 100 accounts at a time and now it seems I can do the whole ~500 I currently have.

When I first got the issue my total was close to 600 and some hadn't been scraped in a week so maybe the issue was in the amount of content to download. I doubt hardware resources was behind the issue though as the VM it runs on is never pegged.
